frontend/src/tests/unit/Api.test.js

```diff
@@ -0,0 +1,51 @@
+import { convertSecondsToStringInfo, getRemainingDate, parseJwt } from '../../utils/Api'
+import { testEndDates } from './testHelpers/endDates'
+import { testInvalidTokens, testProfessorTokens, testStudentTokens } from './testHelpers/tokens'
+
+describe('parseJwt() tests', () => {
+  it.each(testStudentTokens)('student token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testProfessorTokens)('professor token %s parsed correctly', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toEqual(token.token_data)
+  })
+  it.each(testInvalidTokens)('invalid token %s throws error', (token) => {
+    //when
+    const parsedToken = parseJwt(token.access_token)
+
+    //then
+    expect(parsedToken).toBeNull()
+  })
+})
+
+describe('getRemainingDate() tests', () => {
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2022-05-13T12:00:00.000Z'))
+  })
+
+  it.each(testEndDates)('returns correct remaining time to date %s', (endDate) => {
+    //when
+    const remainingDate = getRemainingDate(endDate.date)
+
+    //then
+    expect(remainingDate.asDays()).toEqual(endDate.remainingDays)
```

Comment:
Brakuje mi sprawdzenia dla godzin, minut i sekund. Samo sprawdzenie dni to trochę mało, zawsze coś utnie.

Reply:
To akurat nie jest funkcja która sprawdza ile to jest w dniach, godzinach itd tylko sposób sprawdzenia czy otrzymany okres czasu zgadza się z przewidywanym - to jest dokładny czas w dniach (czyli np równo 36h byłoby 1.5). Nie ma potrzeby tutaj przeliczać tego na sekundy, godziny itd bo to jest ten sam okres czasu wyrażony w różnych jednostkach miary